spec/index/categorical_spec.rb

```diff
@@ -0,0 +1,43 @@
+require 'spec_helper.rb'
+
+describe Daru::CategoricalIndex do
+  let(:idx1) { Daru::CategoricalIndex.new [:a, :b, :a, :a, :c] }
+  let(:idx2) { Daru::CategoricalIndex.new [0, 1, 0, 0, 2] }
+
+  context "#[]" do
+    context "when the category is non-numeric" do
+      context "retrive single category" do
+        subject { idx1[:a] }
+        
+        it { is_expected.to eq Daru::Index.new [0, 2, 3] }
+      end
+      
+      context "retrive multiple categories" do
+        subject {idx1[:a, :c] }
+        
+        it { is_expected.to eq Daru::Index.new [0, 2, 3, 4] }
+      end
+
```

Comment:
Yep, pretty good (it may seem "too complicated" for now, but eventually you'll love it :)).

For simple scenarios, it is **sometimes** acceptable to just `expect( idx1[:a, :c] ).to eq ..`, so you can start the simplest form of specs with it, and then introduce `subject` when there is more than one check. Like "write as less code as possible, but not lesser" :)